cpg_workflows/inputs.py

```diff
@@ -8,41 +8,111 @@
     global _cohort
     if not _cohort:
-        _cohort = create_cohort()
+        _cohort = deprecated_create_cohort()
     return _cohort
 
 
-def create_cohort() -> Cohort:
+def get_multicohort() -> Cohort | MultiCohort:
+    """
+    Return the cohort or multicohort object based on the workflow configuration.
+    """
+    input_datasets = config_retrieve(['workflow', 'input_datasets'], [])
+    custom_cohort_ids = config_retrieve(['workflow', 'input_cohorts'], [])
+    if custom_cohort_ids and input_datasets:
+        raise ValueError('Cannot use both custom_cohort_ids and input_datasets in the same workflow')
+
+    # NOTE: When configuring sgs in the config is deprecated, this will be removed.
+    if config_retrieve(['workflow', 'input_cohorts'], []):
+        if not isinstance(custom_cohort_ids, list):
+            raise ValueError('input_cohorts must be a list')
+        return actual_get_multicohort()
+    return deprecated_get_cohort()
+
+
+def create_multicohort() -> MultiCohort:
+    """
+    Add cohorts in the multicohort.
+    """
+    config = config_retrieve(['workflow'])
+    custom_cohort_ids = config_retrieve(['workflow', 'input_cohorts'], [])
+    multicohort = MultiCohort()
+
+    datasets_by_cohort = get_metamist().get_sgs_for_cohorts(custom_cohort_ids)
+
+    read_pedigree = config.get('read_pedigree', True)
+    for cohort_id in custom_cohort_ids:
+        cohort = multicohort.create_cohort(cohort_id)
+        sgs_by_dataset_for_cohort = datasets_by_cohort[cohort_id]
+        populate_cohort(cohort, sgs_by_dataset_for_cohort, read_pedigree=read_pedigree)
+
+    return multicohort
+
+
+def populate_cohort(cohort: Cohort, sgs_by_dataset_for_cohort, read_pedigree: bool = True):
     """
     Add datasets in the cohort. There exists only one cohort for the workflow run.
     """
+    for dataset_name in sgs_by_dataset_for_cohort.keys():
+        dataset = cohort.create_dataset(dataset_name)
+        sgs = sgs_by_dataset_for_cohort[dataset_name]
+
+        for entry in sgs:
+            metadata = entry.get('meta', {})
+            update_dict(metadata, entry['sample']['participant'].get('meta', {}))
+            # phenotypes are managed badly here, need a cleaner way to get them into the SG
+            update_dict(metadata, {'phenotypes': entry['sample']['participant'].get('phenotypes', {})})
+
+            sequencing_group = dataset.add_sequencing_group(
+                id=str(entry['id']),
+                external_id=str(entry['sample']['externalId']),
+                participant_id=entry['sample']['participant'].get('externalId'),
+                meta=metadata,
+            )
+
+            if reported_sex := entry['sample']['participant'].get('reportedSex'):
+                sequencing_group.pedigree.sex = Sex.parse(reported_sex)
+
+            _populate_alignment_inputs(sequencing_group, entry)
+
+    if not cohort.get_datasets():
+        raise MetamistError('No datasets populated')
```

Comment:
(Dumb question, probably…) Are `skip_sgs`/`only_sgs` being removed once the Old Way is removed?

Reply:
Yes indeed! Once we deprecate the 'Old Way' a user won't need to/be able to specify sgs to skip in prod pipes :)